File: example/c#/CSharpExample/CSharpExample/DotaModule.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using System.IO;
using UnityEngine;
using System;



// dota装备表
public partial class DotaEquipConfig
{

	// 物品ID
	public int id;

	// 物品名称
	public string name;

	// 描述
	public string desc;

	// 售出价格
	public int price;

	// 是否死亡掉落
	public bool isdrop;

	// 攻击力
	public int attack;

	// 合成列表
	public int m_listbornlist;

	// 合成数量
	public int m_listbornnum;
 }

// dota英雄表
public partial class DotaHeroConfig
{

	// 英雄ID
	public int id;

	// 英雄名称
	public string name;

	// 力量
	public double strength;

	// 敏捷
	public double agile;

	// 智力
	public double intelligense;
 }

// dota技能表
public partial class DotaSkillConfig
{

	// 英雄ID
	public int id;

	// 英雄名称
	public string name;

	// 力量
	public double strength;

	// 敏捷
	public double agile;

	// 智力
	public double intelligense;
 }

// dota怪物表
public partial class DotaNpcConfig
{

	// 英雄ID
	public int id;

	// 英雄名称
	public string name;

	// 力量
	public double strength;

	// 敏捷
	public double agile;

	// 智力
	public double intelligense;
 }


public partial class DotaModule
{		
	public static DotaModule instance = new DotaModule();
	
	
	// dota装备表
	public List<DotaEquipConfig> m_listDotaEquipConfig = new List<DotaEquipConfig>();

	// dota英雄表
	public List<DotaHeroConfig> m_listDotaHeroConfig = new List<DotaHeroConfig>();

	// dota技能表
	public List<DotaSkillConfig> m_listDotaSkillConfig = new List<DotaSkillConfig>();

	// dota怪物表
	public List<DotaNpcConfig> m_listDotaNpcConfig = new List<DotaNpcConfig>();

	

	// 物品ID
	public Dictionary<int, DotaEquipConfig> m_mapDotaEquipConfigByid = new Dictionary<int, DotaEquipConfig>();
	
	// 物品名称
	public Dictionary<string, DotaEquipConfig> m_mapDotaEquipConfigByname = new Dictionary<string, DotaEquipConfig>();
	


	// 英雄ID
	public Dictionary<int, DotaHeroConfig> m_mapDotaHeroConfigByid = new Dictionary<int, DotaHeroConfig>();
	
	// 英雄名称
	public Dictionary<string, DotaHeroConfig> m_mapDotaHeroConfigByname = new Dictionary<string, DotaHeroConfig>();
	


	// 英雄ID
	public Dictionary<int, DotaSkillConfig> m_mapDotaSkillConfigByid = new Dictionary<int, DotaSkillConfig>();
	
	// 英雄名称
	public Dictionary<string, DotaSkillConfig> m_mapDotaSkillConfigByname = new Dictionary<string, DotaSkillConfig>();
	


	// 英雄ID
	public Dictionary<int, DotaNpcConfig> m_mapDotaNpcConfigByid = new Dictionary<int, DotaNpcConfig>();
	
	// 英雄名称
	public Dictionary<string, DotaNpcConfig> m_mapDotaNpcConfigByname = new Dictionary<string, DotaNpcConfig>();
	

	
	
	// 物品ID
	public DotaEquipConfig GetDotaEquipConfigByid(int id)
	{
		DotaEquipConfig ret;
		m_mapDotaEquipConfigByid.TryGetValue(id, out ret);
		return ret;
	}
	
	// 物品名称
	public DotaEquipConfig GetDotaEquipConfigByname(string name)
	{
		DotaEquipConfig ret;
		m_mapDotaEquipConfigByname.TryGetValue(name, out ret);
		return ret;
	}
	

	
	// 英雄ID
	public DotaHeroConfig GetDotaHeroConfigByid(int id)
	{
		DotaHeroConfig ret;
		m_mapDotaHeroConfigByid.TryGetValue(id, out ret);
		return ret;
	}
	
	// 英雄名称
	public DotaHeroConfig GetDotaHeroConfigByname(string name)
	{
		DotaHeroConfig ret;
		m_mapDotaHeroConfigByname.TryGetValue(name, out ret);
		return ret;
	}
	

	
	// 英雄ID
	public DotaSkillConfig GetDotaSkillConfigByid(int id)
	{
		DotaSkillConfig ret;
		m_mapDotaSkillConfigByid.TryGetValue(id, out ret);
		return ret;
	}
	
	// 英雄名称
	public DotaSkillConfig GetDotaSkillConfigByname(string name)
	{
		DotaSkillConfig ret;
		m_mapDotaSkillConfigByname.TryGetValue(name, out ret);
		return ret;
	}
	

	
	// 英雄ID
	public DotaNpcConfig GetDotaNpcConfigByid(int id)
	{
		DotaNpcConfig ret;
		m_mapDotaNpcConfigByid.TryGetValue(id, out ret);
		return ret;
	}
	
	// 英雄名称
	public DotaNpcConfig GetDotaNpcConfigByname(string name)
	{
		DotaNpcConfig ret;
		m_mapDotaNpcConfigByname.TryGetValue(name, out ret);
		return ret;
	}
	

	
	// 加载单个文件
	public bool LoadDotaEquipConfig(string basePath)
	{
		string jsonPath = basePath + @"DotaEquipConfig.json";
		string jsonText = File.ReadAllText(jsonPath, System.Text.Encoding.Default);
		
		try
		{
			m_listDotaEquipConfig = (List<DotaEquipConfig>)Newtonsoft.Json.JsonConvert.DeserializeObject(jsonText, typeof(List<DotaEquipConfig>));
		}
		catch (Exception e)
		{
			Debug.Log("加载" + jsonPath + "配置失败，原因 = " + e.Message);
		}			

		for (int i = 0; i < m_listDotaEquipConfig.Count; ++i)
		{
			DotaEquipConfig row = m_listDotaEquipConfig[i];
				m_mapDotaEquipConfigByid[row.id] = row;
				m_mapDotaEquipConfigByname[row.name] = row;
		}
		
		return true;
	}

	// 加载单个文件
	public bool LoadDotaHeroConfig(string basePath)
	{
		string jsonPath = basePath + @"DotaHeroConfig.json";
		string jsonText = File.ReadAllText(jsonPath, System.Text.Encoding.Default);
		
		try
		{
			m_listDotaHeroConfig = (List<DotaHeroConfig>)Newtonsoft.Json.JsonConvert.DeserializeObject(jsonText, typeof(List<DotaHeroConfig>));
		}
		catch (Exception e)
		{
			Debug.Log("加载" + jsonPath + "配置失败，原因 = " + e.Message);
		}			

		for (int i = 0; i < m_listDotaHeroConfig.Count; ++i)
		{
			DotaHeroConfig row = m_listDotaHeroConfig[i];
				m_mapDotaHeroConfigByid[row.id] = row;
				m_mapDotaHeroConfigByname[row.name] = row;
		}
		
		return true;
	}

	// 加载单个文件
	public bool LoadDotaSkillConfig(string basePath)
	{
		string jsonPath = basePath + @"DotaSkillConfig.json";
		string jsonText = File.ReadAllText(jsonPath, System.Text.Encoding.Default);
		
		try
		{
			m_listDotaSkillConfig = (List<DotaSkillConfig>)Newtonsoft.Json.JsonConvert.DeserializeObject(jsonText, typeof(List<DotaSkillConfig>));
		}
		catch (Exception e)
		{
			Debug.Log("加载" + jsonPath + "配置失败，原因 = " + e.Message);
		}			

		for (int i = 0; i < m_listDotaSkillConfig.Count; ++i)
		{
			DotaSkillConfig row = m_listDotaSkillConfig[i];
				m_mapDotaSkillConfigByid[row.id] = row;
				m_mapDotaSkillConfigByname[row.name] = row;
		}
		
		return true;
	}

	// 加载单个文件
	public bool LoadDotaNpcConfig(string basePath)
	{
		string jsonPath = basePath + @"DotaNpcConfig.json";
		string jsonText = File.ReadAllText(jsonPath, System.Text.Encoding.Default);
		
		try
		{
			m_listDotaNpcConfig = (List<DotaNpcConfig>)Newtonsoft.Json.JsonConvert.DeserializeObject(jsonText, typeof(List<DotaNpcConfig>));
		}
		catch (Exception e)
		{
			Debug.Log("加载" + jsonPath + "配置失败，原因 = " + e.Message);
		}			

		for (int i = 0; i < m_listDotaNpcConfig.Count; ++i)
		{
			DotaNpcConfig row = m_listDotaNpcConfig[i];
				m_mapDotaNpcConfigByid[row.id] = row;
				m_mapDotaNpcConfigByname[row.name] = row;
		}
		
		return true;
	}

	
	// 加载配置
	public bool Load(string basePath)
	{
		if (!LoadDotaEquipConfig(basePath))
		{
			return false;
		}
		
		if (!LoadDotaHeroConfig(basePath))
		{
			return false;
		}
		
		if (!LoadDotaSkillConfig(basePath))
		{
			return false;
		}
		
		if (!LoadDotaNpcConfig(basePath))
		{
			return false;
		}
		
		return true;
	}

}






	

File: example/c++/c++_example/cfg/DotaModule.cpp
//------------------------------------------------------------------------------
// 本头文件由工具自动生成，请勿在本文件内做改动
// 日期：
//------------------------------------------------------------------------------

#include "DotaModule.h"
#include <iostream>
#include <rapidjson/document.h>	
#include "tokit_tool.h"




DotaModule DotaModule::instance;





// 物品ID
const DotaEquipConfig* DotaModule::FindDotaEquipConfigByid(int id)
{
	const auto itr = m_mapDotaEquipConfigByid.find(id);
	if (itr == m_mapDotaEquipConfigByid.end())
	{
		return nullptr;
	}

	const DotaEquipConfig* ret = itr->second;
	return ret;
}

// 物品名称
const DotaEquipConfig* DotaModule::FindDotaEquipConfigByname(std::string name)
{
	const auto itr = m_mapDotaEquipConfigByname.find(name);
	if (itr == m_mapDotaEquipConfigByname.end())
	{
		return nullptr;
	}

	const DotaEquipConfig* ret = itr->second;
	return ret;
}




// 英雄ID
const DotaHeroConfig* DotaModule::FindDotaHeroConfigByid(int id)
{
	const auto itr = m_mapDotaHeroConfigByid.find(id);
	if (itr == m_mapDotaHeroConfigByid.end())
	{
		return nullptr;
	}

	const DotaHeroConfig* ret = itr->second;
	return ret;
}

// 英雄名称
const DotaHeroConfig* DotaModule::FindDotaHeroConfigByname(std::string name)
{
	const auto itr = m_mapDotaHeroConfigByname.find(name);
	if (itr == m_mapDotaHeroConfigByname.end())
	{
		return nullptr;
	}

	const DotaHeroConfig* ret = itr->second;
	return ret;
}




// 英雄ID
const DotaSkillConfig* DotaModule::FindDotaSkillConfigByid(int id)
{
	const auto itr = m_mapDotaSkillConfigByid.find(id);
	if (itr == m_mapDotaSkillConfigByid.end())
	{
		return nullptr;
	}

	const DotaSkillConfig* ret = itr->second;
	return ret;
}

// 英雄名称
const DotaSkillConfig* DotaModule::FindDotaSkillConfigByname(std::string name)
{
	const auto itr = m_mapDotaSkillConfigByname.find(name);
	if (itr == m_mapDotaSkillConfigByname.end())
	{
		return nullptr;
	}

	const DotaSkillConfig* ret = itr->second;
	return ret;
}




// 英雄ID
const DotaNpcConfig* DotaModule::FindDotaNpcConfigByid(int id)
{
	const auto itr = m_mapDotaNpcConfigByid.find(id);
	if (itr == m_mapDotaNpcConfigByid.end())
	{
		return nullptr;
	}

	const DotaNpcConfig* ret = itr->second;
	return ret;
}

// 英雄名称
const DotaNpcConfig* DotaModule::FindDotaNpcConfigByname(std::string name)
{
	const auto itr = m_mapDotaNpcConfigByname.find(name);
	if (itr == m_mapDotaNpcConfigByname.end())
	{
		return nullptr;
	}

	const DotaNpcConfig* ret = itr->second;
	return ret;
}




void DotaModule::Clear()
{

	m_vecDotaEquipConfig.clear();

	m_vecDotaHeroConfig.clear();

	m_vecDotaSkillConfig.clear();

	m_vecDotaNpcConfig.clear();


	m_mapDotaEquipConfigByid.clear();
	m_mapDotaEquipConfigByname.clear();

	m_mapDotaHeroConfigByid.clear();
	m_mapDotaHeroConfigByname.clear();

	m_mapDotaSkillConfigByid.clear();
	m_mapDotaSkillConfigByname.clear();

	m_mapDotaNpcConfigByid.clear();
	m_mapDotaNpcConfigByname.clear();


}




bool DotaModule::LoadDotaEquipConfig(std::string basePath)
{
	std::string jsonPath = basePath + "DotaEquipConfig.json";
	std::string jsonText;
	if (!tool::GetFileText(jsonPath.c_str(), jsonText))
	{
		std::cout << "[error] parse failed, not found the file <" << jsonPath << ">!\n";
		return false;
	}
	
    rapidjson::Document doc;
    doc.Parse<0>(jsonText.c_str());
    if (doc.HasParseError()) {
        rapidjson::ParseErrorCode code = doc.GetParseError();
        std::cout << "[error] parse <" << jsonPath << "> failed, error code = " << code << "\n";
        return false;
    }
	
	if (!doc.IsArray())
	{
		std::cout << "[error] parse <" << jsonPath << "> failed, the rapidjson doc is not array, type = " << doc.GetType() << "\n";
		return false;
	}
	
	int n = doc.Size();
	m_vecDotaEquipConfig.resize(n);
	
	for (int i = 0; i < n; ++i)
	{
		// 逐行提取json元素
		rapidjson::Value &v = doc[i];
		
		DotaEquipConfig& row = m_vecDotaEquipConfig[i];

		row.id = v["id"].GetInt();
		
		row.name = v["name"].GetString();
		
		row.desc = v["desc"].GetString();
		
		row.price = v["price"].GetInt();
		
		row.isdrop = v["isdrop"].GetInt();
		
		row.attack = v["attack"].GetInt();
		
		tool::split_str_to_vec<int>(v["bornlist"].GetString(), row.vecbornlist, tool::str_to_int);
		
		tool::split_str_to_vec<int>(v["bornnum"].GetString(), row.vecbornnum, tool::str_to_int);
		
		

		
			m_mapDotaEquipConfigByid[row.id] = &row;
		

		
			m_mapDotaEquipConfigByname[row.name] = &row;
		

		

		

		

		

		

		

	}
	
	return true;
}


bool DotaModule::LoadDotaHeroConfig(std::string basePath)
{
	std::string jsonPath = basePath + "DotaHeroConfig.json";
	std::string jsonText;
	if (!tool::GetFileText(jsonPath.c_str(), jsonText))
	{
		std::cout << "[error] parse failed, not found the file <" << jsonPath << ">!\n";
		return false;
	}
	
    rapidjson::Document doc;
    doc.Parse<0>(jsonText.c_str());
    if (doc.HasParseError()) {
        rapidjson::ParseErrorCode code = doc.GetParseError();
        std::cout << "[error] parse <" << jsonPath << "> failed, error code = " << code << "\n";
        return false;
    }
	
	if (!doc.IsArray())
	{
		std::cout << "[error] parse <" << jsonPath << "> failed, the rapidjson doc is not array, type = " << doc.GetType() << "\n";
		return false;
	}
	
	int n = doc.Size();
	m_vecDotaHeroConfig.resize(n);
	
	for (int i = 0; i < n; ++i)
	{
		// 逐行提取json元素
		rapidjson::Value &v = doc[i];
		
		DotaHeroConfig& row = m_vecDotaHeroConfig[i];

		row.id = v["id"].GetInt();
		
		row.name = v["name"].GetString();
		
		row.strength = v["strength"].GetDouble();
		
		row.agile = v["agile"].GetDouble();
		
		row.intelligense = v["intelligense"].GetDouble();
		
		

		
			m_mapDotaHeroConfigByid[row.id] = &row;
		

		
			m_mapDotaHeroConfigByname[row.name] = &row;
		

		

		

		

	}
	
	return true;
}


bool DotaModule::LoadDotaSkillConfig(std::string basePath)
{
	std::string jsonPath = basePath + "DotaSkillConfig.json";
	std::string jsonText;
	if (!tool::GetFileText(jsonPath.c_str(), jsonText))
	{
		std::cout << "[error] parse failed, not found the file <" << jsonPath << ">!\n";
		return false;
	}
	
    rapidjson::Document doc;
    doc.Parse<0>(jsonText.c_str());
    if (doc.HasParseError()) {
        rapidjson::ParseErrorCode code = doc.GetParseError();
        std::cout << "[error] parse <" << jsonPath << "> failed, error code = " << code << "\n";
        return false;
    }
	
	if (!doc.IsArray())
	{
		std::cout << "[error] parse <" << jsonPath << "> failed, the rapidjson doc is not array, type = " << doc.GetType() << "\n";
		return false;
	}
	
	int n = doc.Size();
	m_vecDotaSkillConfig.resize(n);
	
	for (int i = 0; i < n; ++i)
	{
		// 逐行提取json元素
		rapidjson::Value &v = doc[i];
		
		DotaSkillConfig& row = m_vecDotaSkillConfig[i];

		row.id = v["id"].GetInt();
		
		row.name = v["name"].GetString();
		
		row.strength = v["strength"].GetDouble();
		
		row.agile = v["agile"].GetDouble();
		
		row.intelligense = v["intelligense"].GetDouble();
		
		

		
			m_mapDotaSkillConfigByid[row.id] = &row;
		

		
			m_mapDotaSkillConfigByname[row.name] = &row;
		

		

		

		

	}
	
	return true;
}


bool DotaModule::LoadDotaNpcConfig(std::string basePath)
{
	std::string jsonPath = basePath + "DotaNpcConfig.json";
	std::string jsonText;
	if (!tool::GetFileText(jsonPath.c_str(), jsonText))
	{
		std::cout << "[error] parse failed, not found the file <" << jsonPath << ">!\n";
		return false;
	}
	
    rapidjson::Document doc;
    doc.Parse<0>(jsonText.c_str());
    if (doc.HasParseError()) {
        rapidjson::ParseErrorCode code = doc.GetParseError();
        std::cout << "[error] parse <" << jsonPath << "> failed, error code = " << code << "\n";
        return false;
    }
	
	if (!doc.IsArray())
	{
		std::cout << "[error] parse <" << jsonPath << "> failed, the rapidjson doc is not array, type = " << doc.GetType() << "\n";
		return false;
	}
	
	int n = doc.Size();
	m_vecDotaNpcConfig.resize(n);
	
	for (int i = 0; i < n; ++i)
	{
		// 逐行提取json元素
		rapidjson::Value &v = doc[i];
		
		DotaNpcConfig& row = m_vecDotaNpcConfig[i];

		row.id = v["id"].GetInt();
		
		row.name = v["name"].GetString();
		
		row.strength = v["strength"].GetDouble();
		
		row.agile = v["agile"].GetDouble();
		
		row.intelligense = v["intelligense"].GetDouble();
		
		

		
			m_mapDotaNpcConfigByid[row.id] = &row;
		

		
			m_mapDotaNpcConfigByname[row.name] = &row;
		

		

		

		

	}
	
	return true;
}



bool DotaModule::Load(std::string basePath)
{
	Clear();
		
	if (!LoadDotaEquipConfig(basePath))
	{
		return false;
	}
	
	if (!LoadDotaHeroConfig(basePath))
	{
		return false;
	}
	
	if (!LoadDotaSkillConfig(basePath))
	{
		return false;
	}
	
	if (!LoadDotaNpcConfig(basePath))
	{
		return false;
	}
	
	return true;
}










